session.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"time"
+
+	. "github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+type Client struct {
+	conn      *websocket.Conn
+	broadcast chan *Event
+}
+
+func newClient(conn *websocket.Conn) *Client {
+	return &Client{
+		conn:      conn,
+		broadcast: make(chan *Event),
+	}
+}
+
+// The Dispatcher; we spawn one for every client
+func (c *Client) tx() {
+	defer func() {
```

Comment:
Why is this not just defer { c.conn.Close() }

Why does it have to be its own function and call?